### PR TITLE
match weight tweaks

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [X] MAINT
- [ ] DOC

### Give a brief description for the solution you have provided

Current
image

Updated
image

Updates:
- Fix scale so that the ends of all bars are shown (while keeping match weight of zero in the centre)
- Add dashed lines at +- 10 match weight (similar to parameter estimate chart)

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

**Comment by RossKen** [comment]
Once this has been reviewed I will add to the examples/tutorials/gallery

**Comment by samnlindsay** [comment]
Like the addition of the dotted line to help with scale, thanks.

Given Andy's changes in #1470, this would be a good opportunity to retain a consistent colour scale for match weight, even if we'll move away from red-green in future for accessibility reasons.

Match probability - continuous red-yellow-green from 0 to 1
Match weight - diverging red-grey-green centred on zero

**Comment by RossKen** [comment]
Good point - I can add that in this PR @samnlindsay

**Comment by RossKen** [comment]
Applied @samnlindsay - are you happy with this? Happy to make any other small tweaks here

image

**Comment by ThomasHepworth** [comment]
Do you want me to take a look at the test failures while you're off tomorrow?

**Review by ThomasHepworth — APPROVED** [review]
I guess you could update [this](https://github.com/moj-analytical-services/splink/blob/master/docs/charts/img/match_weights_chart.png), but otherwise I think this is great.

**Comment by ThomasHepworth** [comment]
Failing test - maybe merge master into this branch (I don't think there will be any conflicts) and check if it's been resolved.

File: splink/charts.py [hunk 1]
```diff
@@ -93,6 +96,13 @@ def match_weights_chart(records, as_dict=False):
 
     records = [r for r in records if r["comparison_vector_value"] != -1]
     chart["data"]["values"] = records
+
+    max_value = pd.DataFrame.from_records(records)["log2_bayes_factor"].abs().max()
+    max_value = math.ceil(max_value)
```
[the remark this answers is not in the document]
Comment (ThomasHepworth) [reply splink/charts.py:101]: You could do something like this to resolve the issue?
```py
import pandas as pd
import numpy as np

df = pd.DataFrame({'data': [1, 2, 3, None, 4, 5, np.inf]})
df['data'][df['data'].notna() & (df['data'] != np.inf)].abs().max()
```

There may well be a more performant/better solution using numpy.

[the remark this answers is not in the document]
Comment (RossKen) [reply splink/charts.py:106]: Applied - thanks for the help!

**Pull request merged**